### PR TITLE
br: migrate test-infra to testify for redact

### LINKED ISSUE #28176: migrate test-infra to testify for `br/pkg/redact`

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #28176 .

Signed-off-by: tison <wander4096@gmail.com>

### Release note

```release-note
None
```

/cc @3pointer @kennytm

### CODE REVIEW

File: br/pkg/redact/redact_test.go [hunk 1]
```diff
@@ -3,29 +3,24 @@
...
-type testRedactSuite struct{}
+func TestRedact(t *testing.T) {
+	defer goleak.VerifyNone(t)
 
-func (s *testRedactSuite) SetUpSuite(c *C)    {}
-func (s *testRedactSuite) TearDownSuite(c *C) {}
-
-var _ = Suite(&testRedactSuite{})
-
-func TestT(t *testing.T) {}
```
Comment (tisonkun) [inline br/pkg/redact/redact_test.go:19]: This test doesn't run previously...

**Review by kennytm — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Comment by tisonkun** [comment]
/assign @3pointer @kennytm 

Would you help in merging this PR?

**Pull request merged**